src/PeliasSearchManager.swift

```diff
@@ -125,18 +125,37 @@ public class PeliasResponse: APIResponse {
   let response: NSURLResponse?
   let error: NSError?
   var parsedResponse: PeliasSearchResponse?
+  var parsedError: PeliasError?
   
   init(data: NSData?, response: NSURLResponse?, error: NSError?) {
     self.data = data
     self.response = response
     self.error = error
     if let dictResponse = parseData(data) {
+      if let peliasError = parseErrorFromJSON(dictResponse) {
+        self.parsedError = peliasError
+        return
+      }
       parsedResponse = PeliasSearchResponse(parsedResponse: dictResponse)
     }
   }
   
+  private func parseErrorFromJSON(json: NSDictionary) -> PeliasError? {
+    //First get the error code
+    guard let meta = json["meta"] as? NSDictionary else { return nil }
```

Comment:
I guess the main issue would be if these keys are used in more than one place and were to change it would be nice to update the constant rather than having to search for all there places where it is used. If this is the only place these keys are used then I think this is fine for now.